lib/TokenManager.js

```diff
@@ -13,7 +14,60 @@ var cookieStorage = {
   }
 };
 
-function add(storage, key, token) {
+function clearRefreshTimeout(tokenMgmtRef, key) {
+  clearTimeout(tokenMgmtRef.refreshTimeouts[key]);
+  delete tokenMgmtRef.refreshTimeouts[key];
+}
+
+function clearRefreshTimeoutAll(tokenMgmtRef) {
+  var refreshTimeouts = tokenMgmtRef.refreshTimeouts;
+  for(var key in refreshTimeouts) {
+    if (!refreshTimeouts.hasOwnProperty(key)) {
+      continue;
+    }
+    clearTimeout(refreshTimeouts[key]);
```

Comment:
nit: can we use the clearRefeshTimeout function here so we don't have this logic in two places?